content/hardware/04.pro/boards/portenta-x8/tutorials/16.edge-ai-docker-container/content.md

```diff
@@ -0,0 +1,1145 @@
+---
+title: 'Edge Impulse on Portenta X8 with Docker'
+difficulty: intermediate
+description: 'Learn how to create and deploy an Edge Impulse model on Portenta X8 using Docker containers and a flow sensor for real time anomaly detection.'
+tags:
+  - Portenta X8
+  - Edge Impulse
+  - Docker
+  - Flow Sensor
+  - Sensor Fusion
+author: 'Taddy Ho Chung'
+hardware:
+  - Portenta X8
+  - Flow Sensor
+software:
+  - Edge Impulse
+  - Docker
+  - Python
+  - Arduino IDE
+---
+
+## Overview
+
+Flow sensors play an important role in industrial systems by monitoring fluid movement within pipelines to provide stability and detect anomalies such as leaks, blockages or irregular pressure variations. Traditional flow monitoring methods rely on threshold-based detection, which may fail to capture subtle deviations that indicate potential failures. Real-time pattern classification can improve flow monitoring by integrating machine learning with Edge Impulse, allowing early anomaly detection and predictive maintenance.
+
+![Flow anomaly detection with ML and Arduino Cloud](assets/flow-sense-banner-ref.gif)
+
+This application note shows how to implement a flow anomaly detection system using the Portenta X8, a flow sensor and an Edge Impulse trained machine learning model deployed within a Docker container. The system will capture flow rate data, process it using the trained model and classify flow conditions. 
+
+Classification results can be used for quick decision making, such as triggering an alarm, additional processing based on the available peripherals or forwarding data to Arduino Cloud for remote monitoring and visualization. By leveraging Edge AI capabilities, this approach reduces reliance on cloud processing for anomaly detection while maintaining connectivity for analytics and long-term data tracking.
+
+## Goals
+
+The application note has the following objectives:
+
+- Monitor fluid movement using a flow sensor.
+- Classify flow patterns in real time with a machine learning model trained in Edge Impulse.
+- Deploy the trained model inside a Docker container on the Portenta X8.
+- Share all collected data and classification results to Arduino Cloud for remote monitoring.
+
+## Hardware and Software Requirements
+
+### Hardware Requirements
+
+This application note uses the Portenta X8, integrating a flow sensor for real time fluid monitoring. The required hardware includes:
+
+- [Portenta X8](https://store.arduino.cc/products/portenta-x8) (x1)
+- Portenta Carrier Family: [Hat Carrier](https://store.arduino.cc/products/portenta-hat-carrier) / [Mid Carrier](https://store.arduino.cc/products/portenta-mid-carrier) (x1)
+- Flow Sensor (e.g. YFS201) (x1)
+- [USB-C® cable](https://store.arduino.cc/products/usb-cable2in1-type-c) (x1)
+- Wi-Fi® Access Point or Ethernet with Internet access (x1)
+
+### Software Requirements
```

Comment:
I am missing here the public Edge Impulse project so people can clone it.

Reply:
It is located at the bottom of the list, but I have updated with more explicit description for better representation. Updated!